fix: make namespace parameter optional across all API endpoints

dwmkerr opened the pull request:
## Summary
- Made namespace parameter optional across all ark-api endpoints, defaulting to current context
- Fixed dashboard to not pass namespace in URLs or component props
- Fixed ark-services endpoint to avoid permission issues reading gateway resources

## Changes
- Updated ark-sdk to handle Optional namespace with get_context() fallback
- Modified all 19 ark-api v1 endpoint files to use optional namespace query parameter
- Fixed SecretClient class to properly handle None namespace
- Updated ark-services to use hardcoded port for localhost-gateway (avoids RBAC permission issue)
- Removed namespace from 30+ dashboard components and simplified state management
- Updated dashboard service calls to not pass namespace parameter

## Testing
- Tested all endpoints work without namespace parameter
- Verified ark-services endpoint works without gateway read permissions
- Confirmed secrets endpoint works with updated SDK
- Dashboard components simplified and working

dwmkerr committed:
fix: make namespace parameter optional across all API endpoints

- Update ark-sdk to handle Optional namespace with context fallback
- Update all ark-api v1 endpoints to use optional namespace query parameter
- Fix SecretClient to use get_context() when namespace is None
- Fix ark-services to avoid reading gateway resources (permission issue)
- Remove namespace from dashboard component props and URL paths
- Simplify dashboard to rely on API context instead of passing namespace

av3n93rz commented on services/ark-dashboard/ark-dashboard/lib/services/events.ts:
```diff
@@ -211,9 +211,9 @@ export const eventsService = {
+  async _getEventsForFilters(_namespace: string): Promise<Event[]> {
     try {
-      const result = await this.getAll(namespace, { limit: 200 });
+      const result = await this.getAll(_namespace, { limit: 200 });
```
Seems like we are still using the namespace for the getAll and get functions. Is it still needed here?

av3n93rz commented on services/ark-dashboard/ark-dashboard/components/sections/memory-section.tsx:
```diff
@@ -32,7 +32,7 @@ interface MemorySectionProps {
 
 export function MemorySection({
-  namespace,
+  namespace: _namespace,
```
I think we can remove this since its not used

av3n93rz commented on services/ark-api/ark-api/src/ark_api/api/v1/events.py:
```diff
@@ -66,18 +66,22 @@ async def list_events(
         EventListResponse: List of events in the namespace
     """
+    if namespace is None:
+        from ark_sdk.k8s import get_context
```
Should we move the import to the top of the file?

av3n93rz commented on services/ark-api/ark-api/src/ark_api/api/v1/ark_services.py:
```diff
@@ -147,19 +131,23 @@ async def get_httproutes_for_ark_service(namespace: str, release_name: str) -> L
         ArkServiceListResponse: List of ARK services in the namespace
     """
+    if namespace is None:
+        from ark_sdk.k8s import get_context
```
Should we move this to the top?

av3n93rz commented on lib/ark-sdk/gen_sdk/template/python/base_client.py.template:
```diff
@@ -248,6 +248,9 @@ class ARKResourceClient(Generic[T]):
+
+    def __init__(self, namespace: Optional[str] = None):
+        if namespace is None:
+            from ark_sdk.k8s import get_context
```
Should we move this to the top?

av3n93rz reviewed (changes requested):
Is the namespace selector/creator in the sidebar still works with this?

dwmkerr committed:
chore: move get_context import to top of file

dwmkerr committed:
fix: make namespace optional throughout ARK stack

- Update ark-sdk templates to handle optional namespace with get_context() fallback
- Remove namespace from all 19 ark-api endpoint URLs, make it optional query param
- Remove namespace props from all dashboard components and services
- Fix imports to be at top of files per guidelines
- Update events, secrets, and ark-services to work without explicit namespace
- Let SDK handle namespace detection internally via service account or kubeconfig

dwmkerr committed:
fix: remove all remaining namespace references causing runtime errors

- Remove namespace prop from agent-row.tsx AgentEditor component
- Remove namespace from evaluation-editor.tsx useEffect dependencies
- Remove namespace from events-section.tsx filter string in useEffect

dwmkerr committed:
fix: correct event URL query parameter format

- Change /events& to /events? for proper URL query parameter syntax
- Fixed in agent-row.tsx, model-row.tsx, model-card.tsx, and query page

av3n93rz committed:
fix: remove namespace from tests

av3n93rz committed:
fix: remove unnecessary namespace args

dwmkerr committed:
Merge branch 'main' into fix/namespace-optional-api
